### ISSUE TITLE
Synthesizer model in the case of a custom dataset

### ISSUE THREAD

**Issue by ryanrudes** [NVIDIA/tacotron2#466]
> N.b. When performing Mel-Spectrogram to Audio synthesis, make sure Tacotron 2 and the Mel decoder were trained on the same mel-spectrogram representation.

Does this mean that I need to train separate models if I'm using a custom dataset? Or can I simply train Tacotron using this repo, and use the pretrained synthesizer to convert the mel's to audio? I'm currently experiencing a grinding sound in many cases. While some sentences synthesize decently, some are terribly poor. My dataset has approximately 18000 samples and totals around 30 hours of audio, and I'm 100000 steps into training, but training and validation loss has simply fluctuated around 0.36 for a long time, but quality does still seem to be gradually improving despite no change in validation loss.

I also keep getting this warning when synthesizing:
```
Warning! Reached max decoder steps
```
And each time this appears, the audio is terrible and the alignment is really bad. Every time this warning does not appear, the audio is as expected and the alignment is satisfactory.

Here's some examples of synthesized mels:

**GOOD ALIGNMENT**

<img src="https://i.ibb.co/hZNVjjY/good-alignment.png">

**BAD ALIGNMENT**

<img src="https://i.ibb.co/nwHZ2pX/bad-alignment.png">

As I mentioned previously, is this a product of using a synthesizer that was fine-tuned on a different dataset. If so, how can I fine-tune my own synthesizer?

It also seems to be certain sentences that always come out great, and others always come out poorly. It's most often sentences longer than 5-6 words where this happens most. But I can't see how this could be an overfitting problem considering I have a sufficient amount of data.

**Comment by ryanrudes** [comment]
Nevermind, from a larger window of perspective, this issue changes significantly over the course of longer training. By 150,000 I'd say the problem is half as prominent as it was at 100,000.
